File: packages/backend/src/features/question/model/question.model.ts
import { QuizzemModel } from 'src/core/domain/quizzem.model';
import { GameModel } from 'src/features/game/model/game.model';
import {
  AnswerModeEnum,
  type AnswerMode,
} from 'src/features/question/model/answer-mode';
import { AnswerModel } from 'src/features/question/model/answer.model';
import {
  MediaTypeEnum,
  type MediaType,
} from 'src/features/question/model/media-type';
import { Column, Entity, ManyToMany, OneToMany } from 'typeorm';

@Entity({ name: 'questions' })
export class QuestionModel extends QuizzemModel {
  @Column({ type: 'text' })
  text!: string;

  @Column({
    name: 'answer_mode',
    type: 'enum',
    enum: AnswerModeEnum,
    default: AnswerModeEnum.SINGLE_CHOICE,
  })
  answerMode: AnswerMode = AnswerModeEnum.SINGLE_CHOICE;

  @Column({
    name: 'media_type',
    type: 'enum',
    enum: MediaTypeEnum,
    default: MediaTypeEnum.NONE,
  })
  mediaType: MediaType = MediaTypeEnum.NONE;

  @ManyToMany(() => GameModel, (game) => game.questions)
  games!: GameModel[];

  @OneToMany(() => AnswerModel, (answer) => answer.question, {
    cascade: true,
    orphanedRowAction: 'delete',
  })
  answers!: AnswerModel[];
}

File: packages/backend3/src/main/kotlin/quizzem/backend/features/question/model/QuestionModel.kt
package quizzem.backend.features.question.model

import jakarta.persistence.*
import quizzem.backend.core.domain.QuizzemModel
import quizzem.backend.features.game.model.GameModel

@Entity
@Table(name = "questions")
class QuestionModel(
    @Column(columnDefinition = "TEXT")
    var text: String,

    @Enumerated(EnumType.STRING)
    var answerMode: AnswerMode = AnswerMode.SINGLE_CHOICE,

    @Enumerated(EnumType.STRING)
    var mediaType: MediaType = MediaType.NONE,

    @ManyToMany(
        mappedBy = "questions",
        fetch = FetchType.LAZY
    )
    var games: MutableList<GameModel> = mutableListOf(),

    @OneToMany(
        mappedBy = "question",
        cascade = [CascadeType.ALL],
        orphanRemoval = true,
        fetch = FetchType.LAZY
    )
    var answers: MutableList<AnswerModel> = mutableListOf(),
) : QuizzemModel()
